### ISSUE TITLE
[Feature] 🔐 Encryption at rest — AES-256-GCM encrypt files before upload

### ISSUE THREAD

**Issue by trishantpahwa** [trishantpahwa/dorky#62]
## 🔐 Feature: Encryption at rest — AES-256-GCM encrypt files before upload

![Time: ~1-2 days](https://img.shields.io/badge/Time-~1--2_days-red) ![Difficulty: Advanced](https://img.shields.io/badge/Difficulty-Advanced-critical) ![Language: JavaScript](https://img.shields.io/badge/Language-JavaScript-yellow) ![Flagship feature](https://img.shields.io/badge/Flagship-feature-blueviolet)

> **dorky** is a git-style CLI that keeps your secrets (`.env`, API keys, configs) out of version control by syncing them to AWS S3 or Google Drive — [try it: `npm i -g dorky`](https://npmjs.com/package/dorky) ⭐ **If this project is useful to you, star and fork the repo — it helps a lot!** 🍴

---

### 🎯 The Feature

dorky's files land in S3/Drive **in plaintext** — anyone with bucket/Drive access reads your secrets. The top open [roadmap item](../blob/main/README.md#roadmap) is client-side encryption, so the storage provider only ever sees ciphertext. Node's built-in `crypto` module has everything needed — **zero new dependencies**.

### 📝 Proposed Design (please discuss in comments before building!)

**Opt-in at init:** `dorky --init aws --encrypt` (and an `encrypt` boolean on the MCP `init` tool).

**Key derivation:** passphrase → `crypto.scryptSync(passphrase, salt, 32)`; random 16-byte per-project `salt`. Passphrase comes from `DORKY_PASSPHRASE` env var or an interactive `prompts` password prompt (CLI only; MCP requires the env var).

**Metadata (additive only** — the VS Code extension reads `.dorky/metadata.json`, don't change existing keys**):**
```json
{
  "encryption": { "algorithm": "aes-256-gcm", "salt": "<base64>", "keyCheck": "<base64>" }
}
```
`keyCheck` = encryption of a known constant, so a wrong passphrase fails fast with a clear error instead of writing garbage files.

**File format** (what actually gets uploaded):
```
"DORKY1" magic (6 bytes) | 12-byte IV | 16-byte GCM auth tag | ciphertext
```

**Where it hooks in:**
- `push()`: encrypt the buffer before `PutObjectCommand` / before the Drive `media.body` stream (encrypt to a temp buffer; hashes in metadata stay **plaintext** hashes so change-detection still works). Upload mime becomes `application/octet-stream`. Snapshots to `.dorky-history/` are the same ciphertext — server-side copy still works.
- `pull()` / `checkout()`: after download, detect the `DORKY1` magic bytes → derive key (prompt/env) → decrypt → write plaintext. Files without the magic are written as-is (backwards compatible with projects that pushed before enabling encryption).

### ✅ Acceptance Criteria

- [ ] `--init <provider> --encrypt` stores `encryption` block in metadata; plain init is unchanged
- [ ] Pushed objects are ciphertext with the `DORKY1` header; pull/checkout restore byte-identical plaintext
- [ ] Wrong passphrase → clear error before any file is written
- [ ] Unencrypted legacy projects keep working untouched
- [ ] Unit tests: encrypt/decrypt round-trip, wrong-key failure, magic detection; e2e: full encrypted AWS workflow
- [ ] README section documenting the feature + threat model (protects against storage-provider access; passphrase loss = data loss)
- [ ] `npm test` passes

### 🚀 Quick Info

| | |
|---|---|
| **Difficulty** | Advanced |
| **Time** | ~1–2 days |
| **Files** | `bin/index.js`, `bin/mcp.js` (or `lib/` if #56 landed), `tests/`, `README.md` |
| **Related** | #56 (shared core — strongly recommended to land first), #51 (binary-safe downloads — required, ciphertext is binary!) |

**Please comment before starting** — crypto design deserves a second pair of eyes. 👀

**Questions?** Comment below and we'll help! 🙌 _And if you find dorky useful — ⭐ star and 🍴 fork the repo!_
